### DISCUSSION TITLE
Daemonless Container Image Build

### DISCUSSION THREAD

**Discussion by type-checker** [railwayapp/nixpacks#849] [Ideas]
Is it possible to run Nixpacks with Kaniko instead of BuildKit to build container images? If yes, then how should I change the builder?

CC: @coffee-cup

**Comment by nebulatgs** [comment]
You should be able to have Nixpacks output a Dockerfile and Nix Derivation (using the `-o <PATH>` flag).

This should be something that Kaniko/Podman/alternative builders can ingest.

**Reply by type-checker** [reply]
Hi @nebulatgs , Thanks a lot for your response! That's amazing. It worked but the build speed is very slow. `nix-env -if .nixpacks/nixpkgs-293a28df6d7ff3dec1e61e37cc4ee6e6c0fb0847.nix` takes 40-50 min. to compete the process. So, I thought of using mount prebuilt packages on `nfs` and cache it during the build but I am unable to do the same. If possible, could you plz. guide me on the nfs mount and caching during the build.
> PS: I am running the build in Kaniko pod.
